### DISCUSSION TITLE
Uno 4.7 Templates - how to "opt out" of the new template structure?

### DISCUSSION THREAD

**Discussion by darenm** [unoplatform/uno#11083] [Q&A] [answered]
It has been mentioned that the new template structure with a common library, rather than a shared library, is optional. How do you specify the use of a shared library when creating a new project from the dotnet CLI or the Visual Studio UI?

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the question! It's not optional in the context of creating a new project, but it is when updating existing projects to a newer version of Uno.

If you still want to create a "shared project" template, you can install the dotnet new templates from Uno 4.6:

```
dotnet new uninstall Uno.ProjectTemplates.Dotnet
dotnet new install Uno.ProjectTemplates.Dotnet::4.6.41
```

Then use `dotnet new unoapp` to create a new app.
